consensus/src/quorum_store/quorum_store_wrapper.rs

```diff
@@ -93,20 +97,40 @@ impl QuorumStoreWrapper {
         // TODO: size and unwrap or not?
         let pulled_txns = self
             .mempool_proxy
-            .pull_internal(MAX_SUB_BATCH_SIZE as u64, exclude_txns)
+            .pull_internal(MAX_FRAGMENT_SIZE as u64, exclude_txns)
             .await
             .unwrap();
-        self.batch_in_progress
-            .extend(pulled_txns.iter().map(|txn| TransactionSummary {
-                sender: txn.sender(),
-                sequence_number: txn.sequence_number(),
-            }));
 
-        // TODO: that is a bug! QuorumStore counts in bytes - need to keep track of it.
-        // TODO: also some timer if there are not enough txns
-        if self.batch_in_progress.len() <= self.max_batch_size - MAX_SUB_BATCH_SIZE {
+        let mut end_batch = false;
+        let mut txns_data = Vec::new();
+
+        // TODO: pass TxnData to QuorumStore to save extra serialization.
```

Comment:
This serialization is required once to save to QuorumStore, so we are not adding any additional overhead here once the TODO is taken care of?

Reply:
Not for the serialization.